integration/integration_test.go

```diff
@@ -862,6 +862,11 @@ func (s *IntSuite) TestDisconnectScenarios(c *check.C) {
 		},
 	}
 	for _, tc := range testCases {
+		// TODO: fix the test to work with RecordAtProxy
+		if tc.recordingMode == services.RecordAtProxy {
+			c.Log("Skipping test case with recoridngMode=services.RecodtAtProxy due to https://github.com/gravitational/teleport/issues/3606")
+			continue
+		}
```

Comment:
We should fix this test instead of adding a log line and TODO to fix it before we merge this PR. Otherwise we'll forget and never come back to it.

Reply:
I couldn't figure it out. If you have a quick idea, i'd be happy to take another stab.
I filed a bug, since this is a pre-existing problem that might take some effort to fix.

Reply:
OK, after some digging it turned out trivial - the test didn't enable agent forwarding on the proxy.
Without agent forwarding, proxy recording mode can't work.